### PR TITLE
Add check mode and exit codes

### PR DESCRIPTION
**Pull request by kapsh**

### CODE REVIEW

File: test_docformatter.py [hunk 1]
```diff
@@ -1323,6 +1323,46 @@ def test_standard_in_with_invalid_options(self):
         self.assertIn('cannot be used',
                       process.communicate()[1].decode())
 
+    def test_io_error_exit_code(self):
+        stderr = io.StringIO()
+        ret_code = docformatter._main(
+            argv=['my_fake_program', 'this_file_should_not_exist_please'],
```
Comment (kapsh) [inline test_docformatter.py]: Asking for more robust check.

File: test_docformatter.py [hunk 2]
```diff
@@ -1323,6 +1323,46 @@ def test_standard_in_with_invalid_options(self):
...
+"""
+Print my path and return error code
+""" ''') as filename:
+            stdout = io.StringIO()
+            stderr = io.StringIO()
+            ret_code = docformatter._main(
+                argv=['my_fake_program', '--check', filename],
+                standard_out=stdout, standard_error=stderr, standard_in=None)
+            self.assertEqual(ret_code, 3,
+                             msg='Exit code should be 3')  # FormatResult.check_failed
```
Comment (kapsh) [inline test_docformatter.py]: Didn't use FormatResult directly here for the sake of keeping "contract": readme will say "failed --check returns 3" so it must be 3 if any code in main module changed. But I could be wrong.

**Comment by myint** [comment]
Seems like a reasonable approach to me!

**Comment by kapsh** [comment]
Alright, undrafted it then with a couple of minor changes.

**Pull request merged**